string node_id
string id
